File: src/Programmers/P1/P1_12943.java
package Programmers.P1;

public class P1_12943 {
    public static void main(String[] args) {
        System.out.println(solution(6));
        System.out.println(solution(16));
        System.out.println(solution(626331));
    }

    public static int solution(int num) {
        int tries = 0;
        long numL = num;
        while (numL != 1 && tries < 500) {
            tries++;
            if (numL % 2 == 0)
                numL /= 2;
            else {
                numL *= 3;
                numL++;
            }
        }
        return tries == 500 ? -1 : tries;
    }
}

File: src/Programmers/P1/P1_12943.kt
package Programmers

fun main() {
    println(solution(1))
    println(solution(6))
    println(solution(16))
    println(solution(626331))
}

private fun solution(num: Int): Int {
    var answer = 0
    var temp = num.toLong()
    while (temp != 1L && answer < 500) {
        if (temp % 2 == 0L)
            temp /= 2
        else {
            temp = temp * 3 + 1
        }
        answer++
    }
    return if (answer == 500) -1 else answer
}